Algorithm.CSharp/OpenInterestFuturesRegressionAlgorithm.cs

```diff
@@ -0,0 +1,139 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Algorithm.Framework.Selection;
+using QuantConnect.Data;
+using QuantConnect.Interfaces;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Futures framework algorithm that uses open interest to select the active contract.
+    /// </summary>
+    /// <meta name="tag" content="regression test" />
+    /// <meta name="tag" content="futures" />
+    /// <meta name="tag" content="using data" />
+    /// <meta name="tag" content="filter selection" />
+    public class OpenInterestFuturesRegressionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private static readonly HashSet<DateTime> ExpectedExpiryDates = new HashSet<DateTime>
+        {
+            new DateTime(2013, 10, 29),
+            new DateTime(2013, 11, 26)
+        };
+
+        public override void Initialize()
+        {
+            UniverseSettings.Resolution = Resolution.Minute;
+
+            SetStartDate(2013, 10, 07);
+            SetEndDate(2013, 10, 11);
+            SetCash(10000000);
+
+            // set framework models
+            SetUniverseSelection(
+                new OpenInterestFutureUniverseSelectionModel(
+                    this,
+                    t => new[] {QuantConnect.Symbol.Create(Futures.Metals.Gold, SecurityType.Future, Market.COMEX)},
+                    null,
+                    ExpectedExpiryDates.Count
+                )
+            );
+        }
+
+        public override void OnData(Slice slice)
+        {
+            if (Transactions.OrdersCount == 0 && slice.HasData)
+            {
+                var matched = slice.Keys.Where(s => !ExpectedExpiryDates.Contains(s.ID.Date)).ToList();
+                if (matched.Count != 0)
+                {
+                    throw new Exception("Unexpected symbol(s): " + string.Join(", ", matched));
+                }
+
+                foreach (var symbol in slice.Keys)
+                {
+                    MarketOrder(symbol, 1);
+                }
+            }
+            else if (Portfolio.Any(p => p.Value.Invested))
+            {
+                Liquidate();
+            }
+        }
+
+        /// <summary>
+        ///     This is used by the regression test system to indicate if the open source Lean repository has the required data to
+        ///     run this algorithm.
+        /// </summary>
+        public bool CanRunLocally { get; } = true;
+
+        /// <summary>
+        ///     This is used by the regression test system to indicate which languages this algorithm is written in.
+        /// </summary>
+        public Language[] Languages { get; } = {Language.CSharp, Language.Python};
```

Comment:
nit - `Language.Python` should be removed, else it will try loading the unexisting python version of this algorithm.

Reply:
Ah okay, noted.